src/redisearch_rs/inverted_index/src/tests.rs

```diff
@@ -503,10 +535,55 @@ fn read_using_the_first_block_id_as_the_base() {
 }
 
 #[test]
-#[should_panic(expected = "IndexReader should not be created with an empty block list")]
+fn seeking_records() {
+    // Make two blocks - each with three records
+    let blocks = vec![
+        IndexBlock {
+            buffer: vec![0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 1],
+            num_entries: 3,
+            first_doc_id: 10,
+            last_doc_id: 12,
+        },
+        IndexBlock {
+            buffer: vec![0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 2],
+            num_entries: 3,
+            first_doc_id: 100,
+            last_doc_id: 103,
+        },
+    ];
+
+    let ii = InvertedIndex::from_blocks(IndexFlags_Index_DocIdsOnly, blocks, Dummy);
+    let mut ir = ii.reader();
+
+    let record = ir
+        .seek_record(101)
+        .expect("to be able to read from the buffer")
+        .expect("to get a record");
+
+    assert_eq!(record, RSIndexResult::virt().doc_id(101));
```

Comment:
commenting this out break the test, while it shouldn't, right?

I think there is something wrong in `Decoder::seek()`.  Should it not increment `base` when iterating over records?